### PR TITLE
Bump snappy-java to address CVE

### PR DESCRIPTION
**Pull request by jaguilar-atl**
### Description

Bump snappy-java to address the following potential vulnerabilities.

https://www.cve.org/CVERecord?id=CVE-2023-43642
https://www.cve.org/CVERecord?id=CVE-2023-34453
https://www.cve.org/CVERecord?id=CVE-2023-34454
https://www.cve.org/CVERecord?id=CVE-2023-34455

A similar change has already been pushed into the 2.x branches so this is effectively a backport.
https://github.com/opensearch-project/security/pull/3435

### Testing
Manually built and ran the security plugin according to the [developer guide](https://github.com/opensearch-project/security/blob/1.3/DEVELOPER_GUIDE.md#building).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by willyborankin — APPROVED** [review]

**Review by reta — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -88,6 +88,7 @@ configurations.all {
         force "com.fasterxml.woodstox:woodstox-core:6.4.0"
         force "org.scala-lang:scala-library:2.13.9"
         force "org.apache.bcel:bcel:6.6.0" // This line should be removed once Spotbugs is upgraded to 4.7.4
+        force "org.xerial.snappy:snappy-java:1.1.10.5"
```
Comment (reta) [inline build.gradle:91]: @jaguilar-atl why do we need force here? are there some transitive deps?

Comment (DarshitChanpura) [reply build.gradle:91]: +1

**Comment by willyborankin** [comment]
@jaguilar-atl Could you plz rebase your branch with 1.3?

**Comment by jaguilar-atl** [comment]
> @jaguilar-atl Could you plz rebase your branch with 1.3?

@willyborankin I somehow missed your comment. Apologies for the very late response and thanks to the team for reviewing and merging my PR.

**Comment by willyborankin** [comment]
@jaguilar-atl No problem

**Pull request merged**